Docs/separate readme.mb

jsmith1024 opened the pull request:
More editing will be required. This may be considered a draft.

However, the stated objective has been completed.

Commit:
Separate README.md from home.

Commit:
Set up for refactor.

Commit:
Move documentation files around.

Commit:
Create the needed files for the new documentation structure to work.

Commit:
Finish linking primer.

Commit:
Finish documentation refactor

nekenyu commented on notes/index.html:
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
```
Why is this file present?

nekenyu commented on dox/new_footer.html:
```diff
@@ -12,11 +12,12 @@
 <hr class="footer"/><address class="footer">
-    | <a href="../../home/html/index.html">Home</a>
+    | <a href="index.html">Home</a>
     | <a href="../../refman/html/index.html">User Manual</a>
```
Needs update.

nekenyu commented on dox/new_footer.html:
```diff
@@ -12,11 +12,12 @@
-    | <a href="../../refman/html/index.html">Standard Library Reference</a> |
+    | <a href="Reference.html">Reference Manual</a>
+    | <a href="Primer.html">Bill Primer</a>
+    | <a href="../../refman/html/index.html">Standard Library Reference</a>
```
Needs update.

Commit:
Delete backup file notes/index.html.

nekenyu commented on dox/DoxyHome:
```diff
@@ -58,7 +58,7 @@ PROJECT_LOGO           = "dox/Bill_logo.png"
 # left blank the current directory will be used.
 
-OUTPUT_DIRECTORY       = docs/home
+OUTPUT_DIRECTORY       = docs   # /home
```
Remove # to the end of the line.

nekenyu commented on dox/DoxyHome:
```diff
@@ -753,7 +753,11 @@ WARN_LOGFILE           =
 
-INPUT                  = ./ dox \
+INPUT                  = dox  \
+                         dox/books/reference_manual \
```
This should just be dox and src.  You should not need the subdirectories.

nekenyu commented on dox/DoxyHome:
```diff
@@ -788,8 +792,8 @@ RECURSIVE              = NO
 
-EXCLUDE                = ./test_* \
-                         ~/.. ./.. .. ../.. ../../.. /home/ /home/j/ /home/j/dev
+EXCLUDE                = dox/books/reference_manual/main.dox \
```
Consider dox/books/*/main.dox instead of naming them both.

jsmith1024 replied:
Neat trick!

nekenyu commented on dox/books/Bill_Primer/DoxyLangman:
```diff
@@ -789,7 +789,9 @@ RECURSIVE              = NO
 
 EXCLUDE                = ./test_* \
-                         ~/.. ./.. .. ../.. ../../.. /home/ /home/j/ /home/j/dev
+                         ~/.. ./.. .. ../.. ../../.. \
```
I don't think you need to exclude anything since you have it in one directory.

nekenyu commented on dox/books/reference_manual/DoxyRefman:
```diff
@@ -753,7 +753,9 @@ WARN_LOGFILE           =
 
-INPUT                  = dox/reference_manual \
+INPUT                  = dox/books/reference_manual \
+                         src/compiler \
```
Why are you including the src folders here?

jsmith1024 replied:
Reference manual should be based on the source code.

nekenyu commented on dox/books/reference_manual/DoxyRefman:
```diff
@@ -789,7 +791,9 @@ RECURSIVE              = NO
 EXCLUDE                = ./test_* \
-                         ~/.. ./.. .. ../.. ../../.. /home/ /home/j/ /home/j/dev
+                         ~/.. ./.. .. ../.. ../../.. \
+                         /home/ /home/j/ /home/j/dev \
```
These excludes should not be needed.

Commit:
Clean up doxyfiles and header/footer links.

nekenyu commented on dox/books/Bill_Primer/DoxyLangman:
```diff
@@ -788,10 +788,7 @@ RECURSIVE              = NO
-                         ~/.. ./.. .. ../.. ../../.. \
-                         /home/ /home/j/ /home/j/dev \
-                         dox/books/Bill_Primer/Bill_Primer.dox
+EXCLUDE                = dox/books/Bill_Primer/Bill_Primer.dox
```
I am confused.  Why does language manual have INPUT set to Bill_Primer and exclude Bill_Primer.dox?  Is this a placeholder until you start this book?

nekenyu commented on dox/books/reference_manual/DoxyRefman:
```diff
@@ -790,10 +790,7 @@ RECURSIVE              = NO
-                         ~/.. ./.. .. ../.. ../../.. \
-                         /home/ /home/j/ /home/j/dev \
-                         dox/books/reference_manual/reference.dox
+EXCLUDE                = dox/books/reference_manual/reference.dox
```
I am confused.  Why remove refrence.dox?  Or is it that reference.dox is for html while main.dox is for PDFs?

If that is the case, consider a naming convention such as main.dox (which is a \mainpage) and main_html.dox (which is a \page).

If you are permitted by doxygen, consider main_pdf.dox instead of main.dox.

Commit:
Rename Bill_Primer.dox and reference.dox to main_html.dox.
Rename both corresponding man.dox files to main_pdf.dox.
Adjust all three Doxyfile excludes, accordingly.
